test/test_transforms_v2_refactored.py

```diff
@@ -3859,6 +3859,776 @@ def test_correctness_perspective_bounding_boxes(self, startpoints, endpoints, fo
+    def test_functional(self, make_input):
+        check_functional(F.invert, make_input())
+
+    @pytest.mark.parametrize(
+        ("kernel", "input_type"),
+        [
+            (F.invert_image, torch.Tensor),
+            (F._invert_image_pil, PIL.Image.Image),
+            (F.invert_image, tv_tensors.Image),
+            (F.invert_video, tv_tensors.Video),
+        ],
+    )
+    def test_functional_signature(self, kernel, input_type):
+        check_functional_kernel_signature_match(F.invert, kernel=kernel, input_type=input_type)
+
+    @pytest.mark.parametrize("make_input", [make_image_tensor, make_image_pil, make_image, make_video])
+    def test_transform(self, make_input):
+        check_transform(transforms.RandomInvert(p=1), make_input())
+
+    @pytest.mark.parametrize("fn", [F.invert, transform_cls_to_functional(transforms.RandomInvert, p=1)])
+    def test_correctness_image(self, fn):
+        image = make_image(dtype=torch.uint8, device="cpu")
+
+        actual = fn(image)
+        expected = F.to_image(F.invert(F.to_pil_image(image)))
+
+        assert_equal(actual, expected)
+
+
+class TestPosterize:
+    @pytest.mark.parametrize("dtype", [torch.uint8, torch.float32])
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_kernel_image(self, dtype, device):
+        check_kernel(F.posterize_image, make_image(dtype=dtype, device=device), bits=1)
+
+    def test_kernel_video(self):
+        check_kernel(F.posterize_video, make_video(), bits=1)
+
+    @pytest.mark.parametrize("make_input", [make_image_tensor, make_image, make_image_pil, make_video])
+    def test_functional(self, make_input):
+        check_functional(F.posterize, make_input(), bits=1)
+
+    @pytest.mark.parametrize(
+        ("kernel", "input_type"),
+        [
+            (F.posterize_image, torch.Tensor),
+            (F._posterize_image_pil, PIL.Image.Image),
+            (F.posterize_image, tv_tensors.Image),
+            (F.posterize_video, tv_tensors.Video),
+        ],
+    )
+    def test_functional_signature(self, kernel, input_type):
+        check_functional_kernel_signature_match(F.posterize, kernel=kernel, input_type=input_type)
+
+    @pytest.mark.parametrize("make_input", [make_image_tensor, make_image_pil, make_image, make_video])
+    def test_transform(self, make_input):
+        check_transform(transforms.RandomPosterize(bits=1, p=1), make_input())
+
+    @pytest.mark.parametrize("bits", [1, 4, 8])
+    @pytest.mark.parametrize("fn", [F.posterize, transform_cls_to_functional(transforms.RandomPosterize, p=1)])
+    def test_correctness_image(self, bits, fn):
+        image = make_image(dtype=torch.uint8, device="cpu")
+
+        actual = fn(image, bits=bits)
+        expected = F.to_image(F.posterize(F.to_pil_image(image), bits=bits))
+
+        assert_equal(actual, expected)
+
+
+class TestSolarize:
+    def _make_threshold(self, input, *, factor=0.5):
```

Comment:
`factor` is never set, it doesn't need to be a parameter.

Reply:
Thanks for the heads-up! This is actually a bug. It should have been used in the correctness test. It makes little sense to test thresholds `<=1` for uint8 images.